### PR TITLE
Upgrade LLVM to 10.0.0

### PR DESCRIPTION
**Pull request by mati865**
Fixes https://github.com/msys2/MINGW-packages/issues/3222
Fixes https://github.com/msys2/MINGW-packages/issues/5405
Fixes https://github.com/msys2/MINGW-packages/issues/6126

### CODE REVIEW

File: mingw-w64-clang/PKGBUILD [hunk 1]
```diff
@@ -326,33 +265,34 @@ build() {
...
-    -DLIBUNWIND_ENABLE_SHARED=OFF \
-    -DLLVM_BUILD_STATIC=OFF \
+    -DLIBOMP_ASMFLAGS="${ASM_COMPILER_FLAGS}" \
+    -DLIBOMP_FORTRAN_MODULES=ON \
+    -DLIBUNWIND_ENABLE_SHARED=ON \
+    -DLLDB_RELOCATABLE_PYTHON=ON \
+    -DLLDB_USE_SYSTEM_SIX=ON \
     -DLLVM_BUILD_LLVM_DYLIB=ON \
-    -DLLVM_LINK_LLVM_DYLIB=ON \
+    -DLLVM_BUILD_STATIC=OFF \
```
Comment (ChinouneMehdi) [inline mingw-w64-clang/PKGBUILD:288]: Why would you want to revert #6098 ?

Comment (mati865) [reply mingw-w64-clang/PKGBUILD:288]: To fix https://github.com/msys2/MINGW-packages/issues/6126

Comment (ChinouneMehdi) [reply mingw-w64-clang/PKGBUILD:288]: How do you know that the hangs are because of dynamic linking?

Comment (ChinouneMehdi) [reply mingw-w64-clang/PKGBUILD:288]: LLVM default to ```LLVM_BUILD_STATIC=ON```
Why did you set both ```LLVM_LINK_LLVM_DYLIB``` and ```LLVM_BUILD_STATIC``` to ```OFF``?
I think you should remove these two flags.

Comment (mati865) [reply mingw-w64-clang/PKGBUILD:288]: I haven't modified `LLVM_BUILD_STATIC`, nor #6098 did.
`LLVM_LINK_LLVM_DYLIB` hopefully will be re-enabled someday.

**Comment by Alexpux** [comment]
@mati865  it not buildable
https://gist.github.com/Alexpux/22a7b38bb1e9cf28157e79063297358e

**Comment by mati865** [comment]
It picked headers from `/usr/include`:
```
-- Found LibEdit: D:/msys64/usr/include (found version ".") 
```
Going to investigate shortly.

File: mingw-w64-clang/PKGBUILD (hunk 1 above)
Comment (oscarfv) [reply mingw-w64-clang/PKGBUILD:288]: So LLVM 10 still suffers the problems described in #6098 ?

Comment (mati865) [reply mingw-w64-clang/PKGBUILD:288]: Yeah, LLD keeps hanging in the testsuite.

**Pull request merged**